### DISCUSSION TITLE
Potential enhancements for `persisted.save()`

### DISCUSSION THREAD

**Discussion by richardso21** [olimorris/persisted.nvim#160] [Ideas]
1. By default, session autoload is disabled when `nvim` is executed with arguments or started from stdin. I think it would make sense to also disable saving the session on quit on these circumstances. Otherwise, the existing session save file will be overwritten with the session opened with arguments without warning, which I feel is unintended behavior (or at least this is different than `auto-session`, which I migrated from).
3. When executing `persisted.save()`, it does not respect ignored directories in the config tables. This is because internally it does not double-check `M.allowed_dir()`, even though `autoload` and `load()` does. I believe `save()` should behave similarly, and the `force` option can override this as well.

These issues I could fix by creating my own custom `should_save()` function, but I believe making the changes to the plugin itself would (1) help current and future users, and (2) align the plugin's behavior more closely to what's intended, at least in my opinion. Otherwise, loving this plugin!

**Comment by olimorris** [comment]
Thanks for your comments and taking the time to read the plugin's code as well.

Regarding 1), that is a very good point as that's hugely undesirable. I'll post a fix later today

Regarding 2), `M.allowed_dir()` in the `setup`, `load` and `autoload` functions in turn affects the `VimLeavePre` autocmd which is created in `M.start()`. So if a user is "executing `persisted.save()`" in this case then I would argue their intentions are to overwrite the plugin's default behaviour (as I explain in the README). I appreciate they could also pass in the force flag but this is one of those features where given I don't recall this being raised before, I'm inclined to leave as-is.

**Reply by richardso21** [reply]
Sounds good, thank you so much!
